apps/src/templates/MultipleSectionsAssigner.jsx

```diff
@@ -0,0 +1,324 @@
+    assignmentName: PropTypes.string.isRequired,
+    onClose: PropTypes.func.isRequired,
+    forceReload: PropTypes.bool,
+    courseOfferingId: PropTypes.number,
+    courseVersionId: PropTypes.number,
+    scriptId: PropTypes.number,
+    reassignConfirm: PropTypes.func,
+    isOnCoursePage: PropTypes.bool,
+    isStandAloneUnit: PropTypes.bool,
+    participantAudience: PropTypes.string,
+    // Redux
+    sections: PropTypes.arrayOf(sectionForDropdownShape).isRequired,
+    unassignSection: PropTypes.func.isRequired,
+    assignToSection: PropTypes.func.isRequired,
+    updateHiddenScript: PropTypes.func.isRequired,
+    selectedSectionId: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+
+    let initialSectionsAssignedToCourseList = [];
+
+    // check to see if this is coming from the UNIT landing page - if so add courses featuring this unit
+    if (!this.props.isOnCoursePage) {
+      if (this.props.isStandAloneUnit) {
+        for (let i = 0; i < this.props.sections.length; i++) {
+          if (
+            this.props.courseVersionId ===
+            this.props.sections[i].courseVersionId
+          ) {
+            initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+          }
+        }
+      } else {
+        for (let i = 0; i < this.props.sections.length; i++) {
+          if (this.props.scriptId === this.props.sections[i].unitId) {
+            initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+          }
+        }
+      }
+    } else if (this.props.isOnCoursePage) {
+      // checks to see if this is coming from the COURSE landing page
+      for (let i = 0; i < this.props.sections.length; i++) {
+        if (this.props.courseId === this.props.sections[i].courseId) {
+          initialSectionsAssignedToCourseList.push(this.props.sections[i]);
+        }
+      }
+    }
+
+    this.state = {
+      currentSectionsAssigned: initialSectionsAssignedToCourseList,
+      initialSectionsAssigned: initialSectionsAssignedToCourseList
+    };
+  }
+
+  handleChangedCheckbox = currentSection => {
+    const isUnchecked = this.state.currentSectionsAssigned.some(
+      s => s.code === currentSection.code
+    );
+    if (isUnchecked) {
+      this.setState(state => {
+        let newList = state.currentSectionsAssigned.filter(
+          s => s.code !== currentSection.code
+        );
+        return {currentSectionsAssigned: newList};
+      });
+    } else {
+      this.setState(state => {
+        let newList = [...state.currentSectionsAssigned];
+        newList.push(currentSection);
+        return {currentSectionsAssigned: newList};
+      });
+    }
+  };
+
+  reassignSections = () => {
+    const {
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      scriptId,
+      assignToSection,
+      isOnCoursePage
+    } = this.props;
+    // Assign any courses that need to be assigned
+    for (let i = 0; i < this.state.currentSectionsAssigned.length; i++) {
```

Comment:
nit for the future: I think we generally prefer using `forEach` in these scenarios rather than `for (let i....)`. I don't think it's worth going back and changing these for this PR though.